auth/env_keychain_test.go

```diff
@@ -208,3 +200,76 @@ type providedAuth struct {
 func (p *providedAuth) Authorization() (*authn.AuthConfig, error) {
 	return p.config, nil
 }
+
+func useFallbackKeychain(t *testing.T) {
+	fakeKeychain := FakeKeychain{authMap: map[string]*authn.AuthConfig{
+		"no-env-auth-registry.com": {
+			Auth: "asdf=",
+		},
+	}}
+
+	resolvedKeychain, err := auth.ResolveKeychain(
+		"CNB_REGISTRY_AUTH",
+		[]string{"no-env-auth-registry.com/some-image"},
+		&fakeKeychain,
+	)
+	h.AssertNil(t, err)
+	h.AssertNil(t, os.Unsetenv("CNB_REGISTRY_AUTH")) // unset variable to prove that our test is using a pre-resolved keychain
+
+	registry, err := name.NewRegistry("no-env-auth-registry.com", name.WeakValidation)
+	h.AssertNil(t, err)
+
+	authenticator, err := resolvedKeychain.Resolve(registry)
+	h.AssertNil(t, err)
+
+	header, err := authenticator.Authorization()
+	h.AssertNil(t, err)
+
+	h.AssertEq(t, header, &authn.AuthConfig{Auth: "asdf="})
+}
+
+func fallbackKeychainNoAuthHeader(t *testing.T) {
```

Comment:
```suggestion
func assertDefaultKeychainWithoutHeaderReturnsAnonymous(t *testing.T) {
```
...this function name is starting to feel very long :/

Reply:
I don't have problems with long function names if they make the code more readable